marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.sjs

```diff
@@ -396,10 +396,32 @@ function validateAndRunMapping(mapping, uri) {
   }
 
   const parameterMap = Object.assign({}, {"URI":uri}, userParameterMap);
+  const mappingCount = 1 + (mapping.relatedEntityMappings ? mapping.relatedEntityMappings.length : 0);
```

Comment:
This looks like it can go into a separate method, which will help keep validateAndRunMapping fairly lean and mean. 

Specifically, it looks like you can extract a "validateMappings" function, which makes it very clear that there's a need to validate more than one mapping. And then you can extract a "testMappings" function. Those two functions will reduce the size of validateAndRunMapping - which should likely also be renamed to "validateAndTestMapping" - and make it clear that we first need to validate all the mappings, and then we need to test all the mappings.